resources/scenarios/ln_init.py

```diff
@@ -14,171 +16,391 @@ def add_options(self, parser):
+            thread.start()
+
+        all(thread.join() is None for thread in ch_ann_threads)
+        self.log.info("All LN nodes have complete graph")
+
+        ##
+        # UPDATE CHANNEL POLICIES
+        ##
+        self.log.info("Updating channel policies...")
+
+        def update_policy(self, ln, txid_hex, policy, capacity):
+            self.log.info(f"Sending update from {ln.name} for channel with outpoint: {txid_hex}:0")
+            res = ln.update(txid_hex, policy, capacity)
+            assert (
+                len(res["failed_updates"]) == 0
+            ), f" Failed updates: {res["failed_updates"]}\n txid: {txid_hex}\n policy:{policy}"
+
+        update_threads = []
+        for ch in self.channels:
+            if "source_policy" in ch:
+                ts = threading.Thread(
+                    target=update_policy,
+                    args=(
+                        self,
+                        self.lns[ch["source"]],
+                        ch["txid"],
+                        ch["source_policy"],
+                        ch["capacity"],
+                    ),
+                )
+                ts.start()
+                update_threads.append(ts)
+            if "target_policy" in ch:
+                tt = threading.Thread(
+                    target=update_policy,
+                    args=(
+                        self,
+                        self.lns[ch["target"]],
+                        ch["txid"],
+                        ch["target_policy"],
+                        ch["capacity"],
+                    ),
+                )
+                tt.start()
+                update_threads.append(tt)
+        count = len(update_threads)
+
+        all(thread.join() is None for thread in update_threads)
+        self.log.info(f"Sent {count} channel policy updates")
+
+        self.log.info("Waiting for all channel policy gossip to synchronize...")
+
+        def policy_equal(pol1, pol2, capacity):
+            return pol1.to_lnd_chanpolicy(capacity) == pol2.to_lnd_chanpolicy(capacity)
+
+        def matching_graph(self, expected, ln):
+            while True:
+                actual = ln.graph()["edges"]
+                assert len(expected) == len(actual)
+                done = True
+                for i, actual_ch in enumerate(actual):
+                    expected_ch = expected[i]
+                    capacity = expected_ch["capacity"]
+                    # We assert this because it isn't updated as part of policy.
+                    # If this fails we have a bigger issue
+                    assert int(actual_ch["capacity"]) == capacity
+                    # policy actual/expected source/target
+                    polas = Policy.from_lnd_describegraph(actual_ch["node1_policy"])
+                    polat = Policy.from_lnd_describegraph(actual_ch["node2_policy"])
+                    poles = Policy(**expected_ch["source_policy"])
+                    polet = Policy(**expected_ch["target_policy"])
```

Comment:
It looks like `expected_ch` pulls from the channels config map. I don't think there is enough info in the config map to construct a `Policy`. Would it be sufficient to use the information from the config map and compare it to `actual_ch`?

````json
[{"capacity":50000,"id":{"block":300,"index":2},"push_amt":25000,"target":"tank-0005-ln"}]

````

Reply:
>  I don't think there is enough info in the config map to construct a Policy.

There isn't?

> Would it be sufficient to use the information from the config map and compare it to actual_ch?

The main reason we need `Policy` is to normalize all the terrible naming conventions used by LND. The network.yaml and corresponding channel configmaps conform to the [BOLT](https://github.com/lightningcn/lightning-rfc/blob/master/07-routing-gossip.md#the-channel_update-message) and the `Policy` class is used to translate between (both of) LND's non-spec names and then whatever CLN and Eclair use as well...

Reply:
I may be confused, but I'm seeing this in the channel config map: 

````JSON
[{"capacity":50000,"id":{"block":300,"index":2},"push_amt":25000,"target":"tank-0005-ln"}]
````

And it seems like `Policy` wants this:

````python
        cltv_expiry_delta: int,
        htlc_minimum_msat: int,
        fee_base_msat: int,
        fee_proportional_millionths: int,
        htlc_maximum_msat: int,
````

Reply:
Ah yeah specifying a channel policy is not required, lnd will just use its defaults. So you can have channel config maps without `source_policy` and/or `target_policy`. So probably I need to just not try to check channels without a policy defined at all... In fact if you try to deploy `test/data/ln` the ln_init scenario will fail for this reason! And that isn't covered by a test.
Good catch! thanks.

Reply:
fixed https://github.com/bitcoin-dev-project/warnet/pull/664/commits/ddc93316b350d55dac6a799a2bfaa3beb8005d3f